### PR TITLE
boards: Enable thread local storage for some config

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
to improve the ci coverage

## Impact
Minor

## Testing
Pass CI

### CODE REVIEW

**Comment by pkarashchenko** [comment]
Please fix build errors

**Review by pkarashchenko — APPROVED** [review]
LGTM!

File: boards/arm/kinetis/kwikstik-k40/scripts/kwikstik-k40.ld [hunk 1]
```diff
@@ -86,6 +86,18 @@ SECTIONS
         __exidx_end = ABSOLUTE(.);
     } > progflash
 
+    .tdata : {
```
Comment (pkarashchenko) [inline boards/arm/kinetis/kwikstik-k40/scripts/kwikstik-k40.ld:89]: Need similar fix for RISCV

Comment (xiaoxiang781216) [reply boards/arm/kinetis/kwikstik-k40/scripts/kwikstik-k40.ld:89]: Done.

**Pull request merged**